nilearn/regions/signal_extraction.py

```diff
@@ -101,8 +101,10 @@ def img_to_signals_labels(imgs, labels_img, mask_img=None,
         labels_data[np.logical_not(mask_data)] = background_label
 
     data = _safe_get_data(imgs)
+    target_datatype = np.float64 if data.dtype.kind == 'i' else data.dtype
```

Comment:
this will fail if dtype is bool or unsigned int for example. instead make it float32 if the data is float32 and float64 otherwise

Reply:
If data was not a float64 to begin with, do we need the precision of float64 in target signals?

Reply:
float32 -> float32
anything else -> float64

or we could also consider

float64 -> float64
anything else -> float32

but not the current behaviour which maintains the problem that motivated this PR for anything else than int

Reply:
> the current behaviour which maintains the problem 

If so, then the test I wrote is incorrect. Would you review that please?

Reply:
I believe `dtype.kind='i'` covers unsigned ints.
I should add a test for bool as well.

Reply:
+1 on @jeromedockes 's comment

Reply:
Got it. I'll update this and the test.